flink-core/src/main/java/org/apache/flink/util/CollectionUtil.java

```diff
@@ -56,10 +56,21 @@ private CollectionUtil() {
         throw new AssertionError();
     }
 
+    /** Returns true if the given collection is null or empty. */
     public static boolean isNullOrEmpty(Collection<?> collection) {
         return collection == null || collection.isEmpty();
     }
```

Comment:
the collection can be empty 🔥 

```suggestion
    public static boolean isNullOrEmpty(Collection<?> collection) {
        return collection == null ? false : collection.isEmpty();
    }
```